akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala

```diff
@@ -93,8 +100,15 @@ final class ClusterSettings(val config: Config, val systemName: String) {
 
   val AllowWeaklyUpMembers = cc.getBoolean("allow-weakly-up-members")
 
-  val Team: String = cc.getString("team")
-  val Roles: Set[String] = immutableSeq(cc.getStringList("roles")).toSet + s"team-$Team"
+  val Team: Team = cc.getString("team")
+  val Roles: Set[String] = {
+    val configuredRoles = (immutableSeq(cc.getStringList("roles")).toSet) requiring (
+      _.forall(!_.startsWith(TeamRolePrefix)),
+      s"Roles must not start with '${TeamRolePrefix}' as that is reserved for the cluster team setting"
```

Comment:
Ah, nevermind my comment above then, this does what I asked for there https://github.com/akka/akka/pull/23239/files#r124457953